apps/dashboard-service/src/dashboard/dto/tree-restoration-goal.service.ts

```diff
@@ -0,0 +1,163 @@
+import { Injectable } from "@nestjs/common";
+import { Project, Site, SiteReport, TreeSpecies } from "@terramatch-microservices/database/entities";
+import { DashboardQueryDto } from "./dashboard-query.dto";
+import { DashboardProjectsQueryBuilder } from "../dashboard-query.builder";
+import { Op, fn, col, literal } from "sequelize";
+import { Literal } from "sequelize/types/utils";
+
+interface DateResult {
+  year: number;
+  month: number;
+}
+
+@Injectable()
+export class TreeRestorationGoalService {
+  async getTreeRestorationGoal(query: DashboardQueryDto) {
+    const projectsBuilder = new DashboardProjectsQueryBuilder(Project, [
+      {
+        association: "organisation",
+        attributes: ["uuid", "name", "type"]
+      }
+    ]).queryFilters(query);
+
+    const projects = await projectsBuilder.execute();
+
+    const projectIds = projects.map(project => project.id);
+    const forProfitProjectIds = projects
+      .filter(project => project.organisation?.type === "for-profit-organization")
+      .map(project => project.id);
+
+    const nonProfitProjectIds = projects
+      .filter(project => project.organisation?.type === "non-profit-organization")
+      .map(project => project.id);
+
+    const totalTreesGrownGoal = projects.reduce((sum, project) => sum + (project.treesGrownGoal ?? 0), 0);
```

Comment:
:art: this is fine, but lodash does have a handy utility for this: `sumBy(projects, "treesGrownGoal")`